external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -1427,17 +1434,45 @@ impl SymbolicatorRunner {
                                 }
                                 continue;
                             }
-                            pkgs_to_analyze
-                                .entry(root_dir.unwrap())
-                                .or_insert_with(BTreeSet::new)
-                                .insert(starting_path.clone());
+                            // we want to collect either all modified Move files (as BTreeSet<Some(PathBuf)> or,
+                            // if one of the paths for a package is not a Move file, have aonly one BTreeSet
+                            // None entry. This will be later passed to symbolicator as Some(Vec<PathBuf>) or
+                            // None, to indicate if incremental symbolication is possible or not.
+                            let root_dir = root_dir_opt.unwrap();
+                            if let Some(mut modified_files) = pkgs_to_analyze.remove(&root_dir) {
```

Comment:
Is there a reason to `remove` and `insert` vs simply using `entry` with the appropriate `std::mem::replace`?